Leaderboard: setup project

hamid-murambiwa opened the pull request:
This is a JavaScript project for the Leaderboard list app, using Webpack and ES6 features, notably modules.
Made the following changes:
- Create an npm project with Webpack
- Use ES6 modules, with import and export
- Add HTML file with the page structure

hamid-murambiwa committed:
Made the following changes:
Create an npm project with webpack
Use ES6 modules, with import and export
Add html file with the page structure

hamid-murambiwa committed:
Update linters files

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Update README.md file

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Linters fix

hamid-murambiwa committed:
Bug fix

o-rashad reviewed (changes requested):
Hi @hamid-murambiwa 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a pull request commits the entire `node_modules` directory which makes your pull request unreadable. [LINK](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=Do%20not%20commit%20files%20that%20are%20not%20related%20to%20the%20project.%20Instead%2C%20add%20them%20to%20a%20.gitignore%20file.%20A%20common%20mistake%20is%20to%20commit%20entire%20node_modules%20directory%20which%20makes%20your%20pull%20request%20unreadable.)

This is how it looks like from a Code reviewer perspective :

![29323bb0-5e18-4d37-8a0d-67d3677e912d](https://user-images.githubusercontent.com/24830039/141046798-210d1e41-76f7-4f63-b2c6-d3d0f4c9c02d.png)


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

hamid-murambiwa committed:
Delete node_modules directory

eliasecasta commented on package.json:
```diff
@@ -0,0 +1,47 @@
+  "dependencies": {
+    "express": "^4.17.1",
+    "lodash": "^4.17.21",
+    "webpack": "^5.62.1",
```
- [x] webpack should be present in the devDependencies section of the package.json file, please do so to comply with the project requirements.

eliasecasta commented on src/build.html:
```diff
@@ -0,0 +1,59 @@
+            Refresh
+          </button>
+        </div>
+<ul>
+  <li>
+   <p>Name: 100</p>
+  </li>
+  <li>
+   <p>Name: 100</p>
+  </li>
+  <li>
+    <p>Name: 100</p>
+  </li>
+  <li>
+    <p>Name: 100</p>
+  </li>
+  <li>
+    <p>Name: 100</p>
+  </li>
+  <li>
+    <p>Name: 100</p>
+  </li>
+  <li>
+    <p>Name: 100</p>
+  </li>
+</ul>
```
- [x] This section should be generated using JS DOM, please do so to comply with the project requirements. Don't forget about ES6 syntax.

eliasecasta commented on src/build.html:
```diff
@@ -0,0 +1,59 @@
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>LearderBoard</title>
```
- [x] I can see that you added styling to your app but you forgot to import it in your HTML, please do so to match the wireframe provided by Microverse.

eliasecasta reviewed (changes requested):
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @hamid-murambiwa, I am Elias, your reviewer for this PR

Well done so far with your project!

Your project is almost done, the styling matches the wireframe (I had to manually import it) and your PR details are on point, great job so far!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [ ] Please render the leaderboard with JS instead of HTML to comply with the project requirements.
- [ ] Please import your stylesheet to your HTML so your styling gets applied.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

hamid-murambiwa committed:
Make the following changes:
generated using JS DOM
removed webpack to develop
devDependencies section of the package.json file
import style.css file into your HTML

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Update javascript linters

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Bug fix

hamid-murambiwa committed:
Linters fix

carlos-ssh reviewed (approved):
# 🍾🍾 **STATUS APPROVED** 👏🏽👏🏽

Hi @hamid-murambiwa , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



![Approved](https://user-images.githubusercontent.com/22258274/138466108-6b14c541-6a88-44c5-ba76-c64992ac1835.gif)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
